dashboard/app/helpers/users_helper.rb

```diff
@@ -1,49 +1,90 @@
 module UsersHelper
   include ApplicationHelper
 
-  # Summarize the current user's progress within a certain script.
+  # Summarize a user and his or progress progress within a certain script.
+  # Example return value:
+  # { "linesOfCode": 34, "linesOfCodeText": "Total lines of code: 34", "disableSocialShare": true,
+  #   "levels": {"135": {"status": "perfect", "result": 100}}}
   def summarize_user_progress(script, user = current_user, exclude_level_progress = false)
     user_data = {}
-    uls = {}
-    script_levels = script.script_levels
-    if user
-      lines = user.total_lines
+    merge_user_summary(user_data, user)
+    merge_script_progress(user_data, user, script)
```

Comment:
I think `summarize_user_progress` needs to be passed thru to `merge_script_progress`.